docs/middleware/callbacks/interfaces.md

```diff
@@ -0,0 +1,123 @@
+<!--
+order: 3
+-->
+
+# Interfaces
+
+The callbacks middleware requires certain interfaces to be implemented by the underlying ibc application and the secondary application. If you're simply wiring up the callbacks middleware to an existing ibc application stack such as `icacontroller` and `wasm`, you can skip this section.
+
+## Interfaces for developing the Underlying IBC Application
```

Comment:
Maybe a short intro here about why these 2 interfaces are needed and how they are used together would also help.

Reply:
Which 2 interfaces. I think i tried to add more context into some of them. Also this might be addressed by the upcoming migration docs